### PR TITLE
2.0 API error handling modifications

### PR DESCRIPTION
**Pull request by dalebremner**
@RackHD/corecommitters @BillyAbildgaard @keedya @zyoung51

### CODE REVIEW

**Comment by keedya** [comment]
looks good to me :+1:

File: lib/services/swagger-api-service.js [hunk 1]
```diff
@@ -207,7 +210,12 @@ function swaggerFactory(
             res.status(status).send(data);
         })
         .catch(function(err) {
-            next(_processError(err));
+            if (res.locals.errorStatus) {
+                next();
```
Comment (brianparry) [inline lib/services/swagger-api-service.js]: @dalebremner What are we handling off to from the error handler fitting?  I think the error handler should always close the request in some way, so I would think that calling next will complain that headers have already been sent.  Should we be doing something like calling `res.status(500).json(someDefaultError)`

Comment (brianparry) [inline lib/services/swagger-api-service.js]: So it looks like the default swagger error handler fitting does the same thing: https://github.com/theganyo/swagger-node-runner/blob/master/fittings/json_error_handler.js, so ignore my comment.

**Comment by brianparry** [comment]
:+1:

**Pull request merged**